drivers/flash/flash_simulator.c

```diff
@@ -343,15 +350,23 @@ static void flash_sim_page_layout(struct device *dev,
 }
 #endif
 
+static const struct flash_parameters *
+flash_sim_get_parameters(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return &flash_sim_parameters;
+}
+
 static const struct flash_driver_api flash_sim_api = {
 	.read = flash_sim_read,
 	.write = flash_sim_write,
 	.erase = flash_sim_erase,
 	.write_protection = flash_wp_set,
-	.write_block_size = FLASH_SIMULATOR_PROG_UNIT,
 #ifdef CONFIG_FLASH_PAGE_LAYOUT
 	.page_layout = flash_sim_page_layout,
 #endif
+	.get_parameters = flash_sim_get_parameters,
```

Comment:
Placement of this one isn't the same as the others.

Reply:
Done